scripts/dev/filename-checker

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+#
+# @author muskater
+# @brief Checks that only files exist that conform to the naming convention: [/.a-zA-Z0-9_-]*
+# @date 18.11.2021
+# @tags validation
+
+if [ $# -ne 0 ]; then
+	echo "Usage: filename-checker"
+	exit
+fi
+
+# Check if allow-list exists
+ALLOW_LIST_FILE=./tests/filename.whitelist
+if [ ! -f "$ALLOW_LIST_FILE" ]; then
+	echo >&2 "$ALLOW_LIST_FILE dose not exists."
+	exit 1
+fi
+
+# Store the content of the allow-list into ALLOW_LIST but ignore
+# every line that starts with '#' or spaces
+ALLOW_LIST=$(grep -Ev '^(#|[[:space:]]*$)' "./tests/filename.whitelist")
+
+# find any file that is non-compliant
+# save them to the variable FORBIDDEN_NAMES
+FORBIDDEN_NAMES=$(find . | grep -v '^[-_/.a-zA-Z0-9]*$')
+
+# All file names that offend the rule are saved in band_names
+BANNED_NAMES=$(echo "$FORBIDDEN_NAMES" | while read -r test_line; do
+	# Set the delimiter to newline to compare every line
+	# import the variable test_line as l
+	# compare line by line the allow-list with the read line
+	# if a match is found write that to check
```

Comment:
I do not understand the sentence. Was this comment updated? Probably it should be removed, as nothing is written?